### PR TITLE
Get next overhang

### PR DESCRIPTION
**Pull request by Koeng101**
Added a function called `NextOverhangs` and `NextOverhang`. These can be used to generate the next optimal overhang for a list of overhangs. 

`NextOverhangs` is useful for optimal fragmentation of degenerate coded things (ie, if you're looking to split a gene up into different sections for synthesis and need optimal overhangs)

`NextOverhang` uses `NextOverhangs` and is useful when you're building a new list of optimal overhangs for particular applications. For example, I am using it for optimal overhangs for ligation of barcodes to a I-SceI site (this is also represented in the example for this function.

While Fragment currently fragments on the basis of sequence that is already there, these next 2 functions help if you're trying to fragment something which you have sequence control over.

100% test coverage.

### CODE REVIEW

**Comment by Koeng101** [comment]
@TimothyStiles bump

**Comment by TimothyStiles** [comment]
@Koeng101 checks are good. What's the code coverage?

File: synthesis/fragment/fragment.go [hunk 1]
```diff
@@ -34,6 +35,58 @@ func SetEfficiency(overhangs []string) float64 {
...
+// list, along with their efficiencies. This can be used for more optimal
+// fragmentation of sequences with potential degeneracy.
+func NextOverhangs(currentOverhangs []string) ([]string, []float64) {
+	currentOverhangMap := make(map[string]bool)
+	for _, overhang := range currentOverhangs {
+		currentOverhangMap[overhang] = true
+	}
+	bases := []rune{'A', 'T', 'G', 'C'}
+	var overhangsToTest []string
+	for _, base1 := range bases {
```
Comment (TimothyStiles) [inline synthesis/fragment/fragment.go:48]: okay I have to ask. What's with this super nested control flow? Can this be simplified?

**Review by TimothyStiles — CHANGES_REQUESTED** [review]
@Koeng101

File: synthesis/fragment/fragment.go (hunk 1 above)
Comment (Koeng101) [reply synthesis/fragment/fragment.go:48]: Not sure if simplified, but could probably be smaller by using a recursive function. I think it'd be more complex tho, I think this is reasonable cause you can clearly see what it is doing.

**Comment by Koeng101** [comment]
> @Koeng101 checks are good. What's the code coverage?

100% test coverage.

File: synthesis/fragment/fragment.go (hunk 1 above)
Comment (Koeng101) [reply synthesis/fragment/fragment.go:53]: Comment added.

**Pull request merged**